### PR TITLE
chore: remove gin and zap from test app

### PR DESCRIPTION
**Pull request by silvestre**
# Issue
As an app-autoscaler developer
I want to have as few dependencies as possible
So that I don't have to watch out for so many vulnerabilities in
unmaintained dependencies.

# Fix

- As we already have two HTTP libraries (gorilla/mux and net/http) get
rid of a third one for the test app (gin).
- As we already have two log libraries (lager and slog) get rid of a
third one for the test app (zap).

# AI Disclaimer

Prompted by the committer, most of the work was done by GitHub copilot.

### CODE REVIEW

File: acceptance/assets/app/go_app/internal/app/app.go [hunk 1]
```diff
@@ -1,64 +1,153 @@
...
+						slog.String("path", r.URL.Path),
+						slog.String("method", r.Method),
+					)
+					http.Error(w, "Internal Server Errorf", http.StatusInternalServerError)
+				}
+			}()
+			next.ServeHTTP(w, r)
+		})
+	}
+}
```
Comment (geigerj0) [inline acceptance/assets/app/go_app/internal/app/app.go]: do we really require this?

File: acceptance/assets/app/go_app/internal/app/memory.go [hunk 2]
```diff
@@ -38,61 +37,97 @@ type ListBasedMemoryGobbler struct {
...
-		if memoryTest.IsRunning() {
-			logger.Info("stop mem test")
-			memoryTest.StopTest()
-			logMemoryUsage(logger, "after freeing memory")
-			c.JSON(http.StatusOK, gin.H{"status": "close memory test"})
-		} else {
-			Error(c, http.StatusBadRequest, "memory test not running")
+	mux.HandleFunc("GET /memory/stop", func(w http.ResponseWriter, r *http.Request) {
+		if !memoryTest.IsRunning() {
+			Errorf(logger, w, http.StatusConflict, "memory test is not running")
```
[the remark this answers is not in the document]
Comment (silvestre) [reply acceptance/assets/app/go_app/internal/app/memory.go]: That should be fine, the "app-curler" has `-f`/`--fail` set that should error out on both codes, and `409` fits even better than `400` imo.

File: acceptance/assets/app/go_app/internal/app/disk.go [hunk 3]
```diff
@@ -4,45 +4,46 @@ import (
...
 		duration := time.Duration(minutes) * time.Minute
 		spaceInMB := utilisation * 1000 * 1000
 		if err = diskOccupier.Occupy(spaceInMB, duration); err != nil {
-			Error(c, http.StatusInternalServerError, "error invoking occupation: %s", err.Error())
+			Errorf(logger, w, http.StatusInternalServerError, "error invoking occupation: %s", err.Error())
 			return
 		}
-		c.JSON(http.StatusOK, gin.H{"utilization": utilisation, "minutes": minutes})
+		if err := writeJSON(w, http.StatusOK, JSONResponse{"utilization": utilisation, "minutes": minutes}); err != nil {
+			slog.Error("Failed to write JSON response", slog.Any("error", err))
```
[the remark this answers is not in the document]
Comment (silvestre) [reply acceptance/assets/app/go_app/internal/app/disk.go]: Fixed in https://github.com/cloudfoundry/app-autoscaler/pull/911/changes/56cd5c6c6812fc87cc80d7b73ca3066ff2dda255

File: acceptance/assets/app/go_app/internal/app/memory.go [hunk 4]
```diff
@@ -38,61 +37,61 @@ type ListBasedMemoryGobbler struct {
...
+func parsePositiveInt64(r *http.Request, name string) (int64, error) {
+	val, err := strconv.ParseInt(r.PathValue(name), 10, 64)
+	if err != nil {
+		return 0, fmt.Errorf("invalid %s: %s", name, err.Error())
+	}
+	if val < 1 {
+		return 0, fmt.Errorf("%s must be > 0", name)
+	}
+	return val, nil
+}
```
Comment (geigerj0) [inline acceptance/assets/app/go_app/internal/app/memory.go]: why such an elaborated parsing here that expects a positive int and in the other handlers (e.g. the one for disk) not?

File: acceptance/assets/app/go_app/internal/app/app.go [hunk 5]
```diff
@@ -1,64 +1,62 @@
 package app
 
 import (
+	"encoding/json"
```
Comment (geigerj0) [inline acceptance/assets/app/go_app/internal/app/app.go:4]: something is wrong with the deploy: https://github.com/cloudfoundry/app-autoscaler/actions/runs/24353135289/job/71112796681?pr=911#step:7:768

File: acceptance/assets/app/go_app/internal/app/app.go [hunk 6]
```diff
@@ -1,64 +1,62 @@
...
+	return json.NewEncoder(w).Encode(data)
+}
+
+func respondJSON(logger *slog.Logger, w http.ResponseWriter, data JSONResponse) {
+	if err := writeJSON(w, http.StatusOK, data); err != nil {
+		logger.Error("Failed to write JSON response", slog.Any("error", err))
+	}
+}
+
+func Errorf(logger *slog.Logger, w http.ResponseWriter, statusCode int, format string, args ...any) {
```
Comment (geigerj0) [inline acceptance/assets/app/go_app/internal/app/app.go]: the function name of `Errorf` is a bit hard to understand. how about something that expresses the purpose of the function better (e.g. `writeErrorf`)?

File: acceptance/assets/app/go_app/internal/app/memory.go (hunk 4 above)
Comment (silvestre) [reply acceptance/assets/app/go_app/internal/app/memory.go]: Because SonarQube was complaining about complexity in this handler, but good idea, I'll see about reusing it elsewhere, even if those handlers are not yet too complex.

Comment (silvestre) [reply acceptance/assets/app/go_app/internal/app/memory.go]: done in https://github.com/cloudfoundry/app-autoscaler/pull/911/changes/34922de1686de9d804aeb1280a413a182e4da586

File: acceptance/assets/app/go_app/internal/app/app.go (hunk 6 above)
Comment (silvestre) [reply acceptance/assets/app/go_app/internal/app/app.go]: Naming is hard.

- In https://github.com/cloudfoundry/app-autoscaler/pull/911/commits/c6c49ae8ca47ed091dfe78f473c2a357e18745ef I renamed `Errorf` to `respondWithErrorf`
- In https://github.com/cloudfoundry/app-autoscaler/pull/911/commits/7a9430955ab233470c2fca66fd1e4b6309a5a287 I renamed `respondJSON` to `respondOk` 

I chose the verb `respond` as this is about HTTP responses that are written.
I renamed it to `respondWithErrorf` as `repondErrorf` just sounded wrong to me.
But I rename `respondJSON` to `respondOk` as this is the successful case and the shorter name (without `With`) felt appropriate.

WDYT?

Comment (geigerj0) [reply acceptance/assets/app/go_app/internal/app/app.go]: naming is one of the hardest things! :D 

fine for me!

File: acceptance/assets/app/go_app/Makefile [hunk 7]
```diff
@@ -50,7 +54,7 @@ ${app-fakes-dir} ${app-fakes-files} &: ./go.mod ./go.sum ./internal/generate.go
 #  3. `make go-mod-tidy`
 #  4. Optionally: `make generate-fakes` to update the fakes as well.
 .PHONY: go-mod-tidy
-go-mod-tidy: ./go.mod ./go.sum ${go_deps_without_fakes}
+go-mod-tidy: ./go.mod ./go.sum ${go_deps_without_fakes} ## run go mod tidy
```
[the remark this answers is not in the document]
Comment (geigerj0) [reply acceptance/assets/app/go_app/Makefile:73]: up to you! I don't have hard feelings about this 😇

File: acceptance/assets/app/go_app/internal/app/memory.go (hunk 4 above)
Comment (geigerj0) [reply acceptance/assets/app/go_app/internal/app/memory.go]: shall we add a test in each metric-endpoint-test that checks that we don't pass negative ints?

Comment (silvestre) [reply acceptance/assets/app/go_app/internal/app/memory.go]: Done in https://github.com/cloudfoundry/app-autoscaler/pull/911/changes/ca4f5833348867e5ac1a5d28f1f87245478d3a59

I've also switched the reponsetime `/slow` handler to not accept 0 values as this is what `/fast` handler is for and reused the parse function as well: https://github.com/cloudfoundry/app-autoscaler/pull/911/commits/64c54270da8f13caed82ef65e6563f11ab09656d
